### ISSUE TITLE
[FEATURE REQUEST] Loss function

### ISSUE THREAD

**Issue by LdDl** [LdDl/yolo-go#5]
**Is your feature request related to a problem? Please describe.**
Current trainer - https://github.com/LdDl/yolo-go/blob/master/yolo_trainer.go#L11
Current loss function for YOLO bounding boxes - https://github.com/LdDl/yolo-go/blob/master/yolo_trainer.go#L100
I have doubts that it is right code (or even approach). Need to implement it step-by-step and compare with classic C-based implementation (may be take inspirations by some tensorflow/pytorch implementation)

**Describe the solution you'd like and provide pseudocode examples if you can**
According to source code of YOLO:
IOU's - https://github.com/AlexeyAB/darknet/blob/master/src/yolo_layer.c#L166:
```C
float delta_yolo_box(box truth, float *x, float *biases, int n, int index, int i, int j, int lw, int lh, int w, int h, float *delta, float scale, int stride)
{
    box pred = get_yolo_box(x, biases, n, index, i, j, lw, lh, w, h, stride);
    float iou = box_iou(pred, truth);

    float tx = (truth.x*lw - i);
    float ty = (truth.y*lh - j);
    float tw = log(truth.w*w / biases[2*n]);
    float th = log(truth.h*h / biases[2*n + 1]);

    delta[index + 0*stride] = scale * (tx - x[index + 0*stride]);
    delta[index + 1*stride] = scale * (ty - x[index + 1*stride]);
    delta[index + 2*stride] = scale * (tw - x[index + 2*stride]);
    delta[index + 3*stride] = scale * (th - x[index + 3*stride]);
    return iou;
}
```

classes - https://github.com/AlexeyAB/darknet/blob/master/src/yolo_layer.c#L276:
```C
void delta_yolo_class(float *output, float *delta, int index, int class, int classes, int stride, float *avg_cat)
{
    int n;
    if (delta[index]){
        delta[index + stride*class] = 1 - output[index + stride*class];
        if(avg_cat) *avg_cat += output[index + stride*class];
        return;
    }
    for(n = 0; n < classes; ++n){
        delta[index + stride*n] = ((n == class)?1 : 0) - output[index + stride*n];
        if(n == class && avg_cat) *avg_cat += output[index + stride*n];
    }
}
```

**Describe alternatives you've considered and provide pseudocode examples if you can**
nope

**Additional context**
nope
